Sources/MapboxCoreNavigation/RouteOptions.swift

```diff
@@ -47,9 +49,9 @@ extension RouteOptions {
      - parameter waypoint: the Waypoint to exclude.
      - returns: a copy of self excluding the specified waypoint.
      */
-    public func without(_ waypoint: Waypoint) -> RouteOptions {
+    public func without(_ waypoint: Waypoint) -> DirectionsOptions {
```

Comment:
Good catch – the original intention of this method was to return an object of the same type as the receiver, hence the simple “without” naming. It would be better to define a similar but separate `MatchOptions.without(_:)`.

(Swift does allow overloading the method with the same signature except for a different return value. It’s frowned upon, but for example the Polyline package makes use of that language feature.)